src/ImageSharp/Memory/DiscontiguousBuffers/MemoryGroupEnumerator{T}.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+
+namespace SixLabors.ImageSharp.Memory
+{
+    /// <summary>
+    /// A value-type enumerator for <see cref="MemoryGroup{T}"/> instances.
+    /// </summary>
+    /// <typeparam name="T">The element type.</typeparam>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public ref struct MemoryGroupEnumerator<T>
```

Comment:
Yeah, it was mostly a precaution to prevent users from doing stuff like storing the iterator as a field in a class for later use, or other weird things. This type should really only ever be used implicitly by the compiler when using `foreach`, it should be completely transparent to users 😄